07.ls_object/lib/display.rb

```diff
@@ -0,0 +1,148 @@
+# frozen_string_literal: true
+
+require_relative 'file_data'
+COLUMN_NUMBER = 3
+SHORT_LIST_PADDING = 7
+FILETYPES = {
+  'fifo' => 'p',
+  'characterSpecial' => 'c',
+  'directory' => 'd',
+  'blockSpecial' => 'b',
+  'file' => '_',
+  'link' => 'l',
+  'socket' => 's'
+}.freeze
+MODE_TABLE = {
+  '0' => '---',
+  '1' => '--x',
+  '2' => '-w-',
+  '3' => '-wx',
+  '4' => 'r--',
+  '5' => 'r-x',
+  '6' => 'rw-',
+  '7' => 'rwx'
+}.freeze
+class Display
+  def initialize(paths)
+    @files = paths.map { |path| FileData.new(path) }
+  end
+
+  def render_short_list
+    width = max_base_name_width + SHORT_LIST_PADDING
+    formated_files = base_names.map { |file| file.ljust(width) }
+    file_count = count_files
+    row_number = (file_count / COLUMN_NUMBER.to_f).ceil
+    (COLUMN_NUMBER - (file_count % COLUMN_NUMBER)).times { formated_files << '' } if file_count % COLUMN_NUMBER != 0
+    render_lines(formated_files, row_number)
+  end
+
+  def render_long_list
+    max_widths = find_max_widths
+    total = "total #{total_blocks}"
+    render_files = @files.map { |file| build_data(file, max_widths) }.map { |file| format_row(file) }
```

Comment:
```suggestion
    render_files = @files.map { |file| build_data(file, max_widths) }.map { |file| format_row(file) }
```

`|file|` が2回出てきますが、異なるデータに同じ名前を付けるのは混乱の元です。

Reply:
```rb
long_list = @file_statuses.map { |status| organize(status, max_widths) }.map { |organized_status| format_row(organized_status) }
```
FileDataクラスをFileStatusクラスに変更したので、最初の引数にstatus、2回目はorganizeメソッドを通過しているのでorganized_statusに変更しました。
やや安直かなとは思うのですが、他のメソッド名やクラス名の変更に合わせて修正しました。